File: ruby/majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
  res = []
  sorted_arr = sort(nums)

  # puts sorted_arr

  # nums.length.times do |i|
  #   puts "#{nums[i]} == #{nums[i + 1]}"
  #   res << nums if nums[i] == nums[i + 1]
  # end
end

def sort(nums)
  nums.length.times do |i|
    next unless nums[i] && nums[i + 1]

    if nums[i] > nums[i + 1]
      nums[i] = nums[i + 1]
      nums[i + 1] = nums[i]
    end
  end

  puts nums
end

majority_element([2, 2, 1, 1, 1, 2, 2])

File: js/majorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  nums = nums.sort();
  for (let i = nums.length / 2; i--; ) {
    console.log(nums[i]);
  }
};

console.log(majorityElement([2, 2, 1, 1, 1, 2, 2]));
// console.log(majorityElement([3, 2, 3]));
